Improve Linux memory collection with AVLM

mczap opened the pull request:
Linpmem read /proc/kcore during my tests, but never completed the dump, but filled the partition. Tested with CentOS, Ubuntu and RHEL VMs.

Of all the tools I looked at as replacement, AVML proved to be the easiest to use. Compiled as described, on an Ubuntu 18.04.5 LTS VM. The binary should run on many systems and kernel versions.
https://github.com/microsoft/avml#building-on-ubuntu
https://github.com/microsoft/avml#tested-distributions

diogo-fernan commented on nix/tools-nix/mem/readme.md:
```diff
@@ -0,0 +1,3 @@
+AVML compiled as described, on an Ubuntu 18.04.5 LTS VM. The binary should run on many systems and kernel versions.
```
Perhaps this whole description can rather go into the main `README.md` file?

diogo-fernan commented on README.md:
```diff
@@ -202,7 +202,7 @@ registry-system=true
 ## Unix
 
-* **linpmem-2.1.post4** (v2.1.post4): the [Pmem](https://github.com/google/rekall "Rekall GitHub Repository") suite is part of the open-source Recall memory analysis framework, used by *ir-rescue-nix* to dump the memory.
+* **AVML-0.21** (Acquire Volatile Memory for Linux): [AVML](https://github.com/microsoft/avml) is an X86_64 userland volatile memory acquisition tool written in Rust, intended to be deployed as a static binary. AVML can be used to acquire memory without knowing the target OS distribution or kernel a priori. No on-target compilation or fingerprinting is needed.
```
Here the building information can be appended.

diogo-fernan reviewed (changes requested):
Would you mind merging the memory-specific `README.md` file with the generic one?

I can update the change history and the configuration file afterwards to reflect these changes.

mczap commented:
Thank you for your quick reaction and open-mindedness towards my suggestions. I have changed the [readme.md](https://github.com/mczap/ir-rescue/blob/memory-collection-AVML/README.md) as you recommended it and deleted the superfluous one under ./nix/tools-nix/mem/.
Now I have to find out how to set the execution bit via git and your script will be even easier to use. Thanks for your development work

mczap commented:
I have done all the needed changes, is it OK for you to merge it, or would you like further changes?

mczap commented:
On RHEL7 there are still problems with archiving after the collection. In the script there is also a reference to the aff4 file, witch I missed. Please don't merge yet. I try to solve these points first.

diogo-fernan commented:
Cool, thanks for contributing! Merging now.
